src/nnet3/nnet-utils.h

```diff
@@ -53,6 +53,19 @@ int32 NumOutputNodes(const Nnet &nnet);
 /// returns the number of input nodes of this nnet.
 int32 NumInputNodes(const Nnet &nnet);
 
+/// This function works out the interval at which a particular named
+/// input (e.g. input_name == "input" or input_name == "ivector") is
+/// used.  If the input is not used at all this function returns -1;
+/// if it is potentially used on every frame (every "t" value) it returns 1
+/// if it is only used on frame 0 it returns 0; and if it is used only
+/// on multiples of n frames with n > 1, it returns n.
+int32 GetInputInterval(const Nnet &nnet, std::string input_name);
+
+/// internal function called in GetInputInterval().
+int32 GetInputIntervalInternal(const GeneralDescriptor &gen_desc,               
```

Comment:
OK, then it should be named differently, such as GetDescriptorInputInterval.
And as I said previously, at this level you shouldn't be passing in the
vector of node names and the name of the input node you want; you should be
passing in the integer node name corresponding to the input you want.

On Fri, Mar 25, 2016 at 9:10 PM, Yiming Wang <notifications@github.com>
wrote:

> In src/nnet3/nnet-utils.h
> <https://github.com/kaldi-asr/kaldi/pull/468#discussion_r57506274>:
>
> > @@ -53,6 +53,19 @@ int32 NumOutputNodes(const Nnet &nnet);
> >  /// returns the number of input nodes of this nnet.
> >  int32 NumInputNodes(const Nnet &nnet);
> >
> > +/// This function works out the interval at which a particular named
> > +/// input (e.g. input_name == "input" or input_name == "ivector") is
> > +/// used.  If the input is not used at all this function returns -1;
> > +/// if it is potentially used on every frame (every "t" value) it returns 1
> > +/// if it is only used on frame 0 it returns 0; and if it is used only
> > +/// on multiples of n frames with n > 1, it returns n.
> > +int32 GetInputInterval(const Nnet &nnet, std::string input_name);
> > +
> > +/// internal function called in GetInputInterval().
> > +int32 GetInputIntervalInternal(const GeneralDescriptor &gen_desc,
>
> In nnet-descriptor-test.cc, I call GetInputIntervalInternal(), rather than
> GetInputInterval() for testing, (because GetInputInterval() requires nnet
> as an input argument) . So I declare GetInputIntervalInternal() here for
> nnet-desciptor-test.cc to include.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/468/files/a4f4f4761a8997f35e0f6be3759f76ad57ed141c#r57506274>
>